app/script/util/SanitizationUtil.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.util = z.util || {};
+
+z.util.SanitizationUtil = (() => {
```

Comment:
The `SanitizationUtil` uses this structure:

```js
z.util.SanitizationUtil = (() => {
  const _escapeString = string => _.escape(string);
  return {
    escapeString: _escapeString,
  }
}
```

Which can be written much simpler:

```js
z.util.SanitizationUtil = {
  escapeString: (string) => _.escape(string)
}
```

Reply:
The pattern shown above will be a little complicated for methods which use other methods of the same object (like `_getEscapedSelfName`) but therefor we can use a class pattern with static methods.

Reply:
I would say that a `class` for that is too powerful (it allows its consumer to instantiate it, but we don't want that kind of power).
If we don't want something to be instantiated it's better if it is not possible to instantiate it by design (rather than by convention). 
To me, a simple JavaScript object is the perfect abstraction for this util

Reply:
To me, this is a good application of the `rule of least power`. (here it is about choosing the right language for the right job, but we can easily extend it to choosing the right abstraction).
https://www.w3.org/2001/tag/doc/leastPower.html

Reply:
**Very good reasoning with facts!** I would be very happy if we can reduce private functions (starting with the underscore pattern) to a minimum because this is what bothers me the most with the current pattern. Do you have an idea how we can achieve that by keeping the rule of least power?

Reply:
Yes, I might have a solution, thanks for asking :)
I believe here we actually don't need the `_` notation!!!

Let me explain. The `_` notation is here to say to consumer of our class "you should not use this method, please".
Here, because those functions are in an isolated scope (they are only published to the consumer if we decide to expose them), we actually don't need this notation. I would go like this:

```
z.util.SanitizationUtil = (() => {
  // here are the functions definition
  const escapeString = string => _.escape(string);
  return {
     // here you decide what API you publish
    escapeString
  }
}
```

Reply:
Which is basically how it is written now, except for the underscores (which i wouldn't miss at all) :)